### DISCUSSION TITLE
Built-in Next.js Tags not Working

### DISCUSSION THREAD

**Discussion by skylerknight** [markdoc/markdoc#59] [Q&A] [answered]
Anyone else having troubles getting the built-in next.js tags to work?

I'm using the default setup, and exporting the built-in tags from `@markdoc/next.js` as explained in the docs [here](https://markdoc.io/docs/nextjs#built-in-next.js-tags):

```js
// markdoc/tags/Next.markdoc.js

export { comment, head, image, link, script } from '@markdoc/next.js/tags';
```

But they don't seem to work.

**Comment by mfix-stripe** [comment, marked as answer]
@skylerknight could you share a reproduction of your issue, and what in particular you are running into? 

We are using tags from `@markdoc/next.js/tags` here (https://github.com/markdoc/docs/blob/main/markdoc/tags/index.js#L4), which are working.

**Reply by skylerknight** [reply]
Sure no problem. After double-checking, the comment, and link tags work properly. I'm only getting errors with the head and image tags.


As for the image tag, It looks like the built-in `image` tag for next.js only exists in the documentation code snippet [here](https://markdoc.io/docs/nextjs#built-in-next.js-tags). I'm not seeing in the code of `@markdoc/next.js/tags.js`.

As for the head tag, I'm getting the following error after apply this code:

```js
// markdoc/tags/index.js

export { comment, link, head } from "@markdoc/next.js/tags";
```

```js
// pages/index.md

{% head %}
<title>Custom Title</title>
{% /head %}
```

```js
// Console Error

Warning: next-head-count is missing. https://nextjs.org/docs/messages/next-head-count-missing
```

**Reply by mfix-stripe** [reply]
> As for the image tag, It looks like the built-in image tag for next.js only exists in the documentation code snippet [here](https://markdoc.io/docs/nextjs#built-in-next.js-tags).
Ahh, good catch. We originally had an image component, but we decided to remove it since it doesn't work with `next export` and requires some custom finagling. We will reintroduce it if there is better support for `next/image` w/ `next export.

**Reply by mfix-stripe** [reply]
For the `head` component, you need other custom tags for the children, depending on what you want rendered. 

For your example above, you should create a `title` tag like:
```js
// markdoc/tags.js

import {Title} from '../components'

export const title = {
  render: Title
}
```

and in your Markdoc use:

```js
// pages/index.md

{% head %}

{% title %}
Custom Title
{% /title %}

{% /head %}
```

We decided not to include tags like `title` and `meta` in this package to start since they don't relate specifically to Next.js, but we will consider adding them in the future.

**Reply by skylerknight** [reply]
Aha, alright that makes sense.

Does this mean that regular HTML content cannot be parsed with markdoc, but only custom tags?

**Reply by mfix-stripe** [reply]
> Does this mean that regular HTML content cannot be parsed with markdoc, but only custom tags?

At the moment, yes. See https://github.com/markdoc/markdoc/issues/10 for more details. 🙂

**Reply by skylerknight** [reply]
Got it. Thanks!

**Comment by arranf** [comment]
@mfix-stripe I get the following error but when silenced with a `@ts-ignore` it works at runtime. 
`Cannot find module '@markdoc/next.js/tags' or its corresponding type declarations.ts(2307)`

**Reply by mfix-stripe** [reply]
@arranf yes this module doesn't have TypeScript definitions yet, so you would have to write your own. The tags themselves are very basic, so you can also just copy in the ones you want: https://github.com/markdoc/next.js/blob/main/src/tags.js

That TypeScript error will not affect runtime, which it seems @skylerknight is running into.

**Reply by arranf** [reply]
That makes sense - not concerned about the lack of TS definitions. Just thought I'd mention it in case it was the error being experienced!
